soulmask/egg-soulmask.json

```
@@ -0,0 +1,132 @@
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "App ID",
+            "description": "",
+            "env_variable": "SRCDS_APPID",
+            "default_value": "3017300",
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|string|in:3017300",
+            "field_type": "text"
+        },
+        {
+            "name": "Auto Update",
+            "description": "Auto update the server on start",
+            "env_variable": "AUTO_UPDATE",
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        },
+        {
+            "name": "Server Name",
+            "description": "Specifies the name of the game instance displayed in the server list",
+            "env_variable": "SERVER_NAME",
+            "default_value": "Soulmask Server",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:85",
+            "field_type": "text"
+        },
+        {
+            "name": "Map Players",
+            "description": "Specifies the maximum number of players the game instance can support.",
+            "env_variable": "MAX_PLAYERS",
+            "default_value": "60",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric|between:1,70",
+            "field_type": "text"
+        },
+        {
+            "name": "Backup Interval",
+            "description": "Specifies the interval for writing the game database to disk (unit: seconds).",
+            "env_variable": "BACKUP_TIME",
+            "default_value": "960",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric",
+            "field_type": "text"
+        },
+        {
+            "name": "Save Time Interval",
+            "description": "Specifies the interval for writing game objects to the database (unit: seconds).",
+            "env_variable": "SAVE_TIME",
+            "default_value": "600",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric",
+            "field_type": "text"
+        },
+        {
+            "name": "Query Port",
+            "description": "Specifies the Steam query port, UDP, needs to be open to the public.",
+            "env_variable": "QUERY_PORT",
+            "default_value": "27015",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|numeric|between:1024,65536",
+            "field_type": "text"
+        },
+        {
+            "name": "Echo Port",
+            "description": "Maintenance port, used for local telnet server maintenance, TCP, does not need to be open.",
+            "env_variable": "ECHO_PORT",
```

This is more opinionated, but I went with "maintenance port" in my own egg because it makes it clearer how the port is to be used. Not sure if that vibes with you or not.

```suggestion
            "name": "Maintenance Port",
            "description": "Maintenance port, used for local telnet server maintenance, TCP, does not need to be open.",
            "env_variable": "ECHO_PORT",
```